### PR TITLE
Add sas token endpoint

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-869

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/SasTokenControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import java.time.OffsetDateTime;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringJUnitWebConfig
+@SpringBootTest
+@AutoConfigureMockMvc
```
Comment (doncem) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/SasTokenControllerTest.java]: same here

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/SasTokenControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+        assertThat(queryParams.get("sig")).isNotNull();//this is a generated hash of the resource string
+        assertThat(queryParams.get("sv")).contains("2019-02-02");//azure api version is latest
+
+        OffsetDateTime expiresAt = OffsetDateTime.parse(queryParams.get("se")[0]); //expiry datetime for the signature
+        OffsetDateTime now = OffsetDateTime.now();
+        assertThat(expiresAt).isBetween(now, now.plusSeconds(300));
+        assertThat(expiresAt).isBefore(now.plusSeconds(301));
+
+        assertThat(queryParams.get("sp")).contains("wl");//access permissions(write-w,list-l)
```
Comment (doncem) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/SasTokenControllerTest.java]: Same is done in unit test. Perhaps we could go with just "verification" query params parsed successfully and single assertion made for let's say "most significant" parameter?

Comment (aliveni) [reply src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/SasTokenControllerTest.java]: Updated to verify just the "sig" parameter

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/resources/application.conf [hunk 3]
```diff
@@ -0,0 +1,5 @@
+storage.account-name: bulkscan
+storage.account-key: testkey
+
+service.storage-config[0].name: bulkscan
+service.storage-config[0].sas-validity: 300
```
[the remark this answers is not in the document]
Comment (aliveni) [reply src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/resources/application.conf]: done

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/SasTokenControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    @Test
+    @Disabled // TODO Disabled until Exception handler is added
+    public void should_throw_exception_when_service_is_not_configured() throws Exception {
+        MvcResult result = mockMvc
+            .perform(get("/token/unconfiguredservcie"))
```
[the remark this answers is not in the document]
Comment (aliveni) [reply src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/SasTokenControllerTest.java]: changed it as suggested

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/GetWelcomeTest.java [hunk 5]
```diff
@@ -3,15 +3,19 @@
...
 import org.springframework.test.web.servlet.MvcResult;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
-@WebMvcTest
+@SpringJUnitWebConfig
+@SpringBootTest
+@AutoConfigureMockMvc
```
[the remark this answers is not in the document]
Comment (aliveni) [reply src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/GetWelcomeTest.java]: Done

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/SasTokenControllerTest.java (hunk 1 above)
Comment (aliveni) [reply src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/SasTokenControllerTest.java]: Can't apply the same suggestion here. As this test is failing to initiate service class with `WebMvcTest` annotation

**Pull request merged**